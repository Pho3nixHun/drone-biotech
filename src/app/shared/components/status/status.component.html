<p class="font-medium rounded-full w-max px-3 py-1 flex items-center">
    <ng-content />
</p>
