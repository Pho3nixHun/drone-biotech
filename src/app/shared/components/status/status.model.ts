export interface StatusVM {
    styles: string;
}
