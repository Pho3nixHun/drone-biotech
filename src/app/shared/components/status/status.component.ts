import { Component, input } from '@angular/core';
import { StatusVM } from './status.model';
import { NgClass } from '@angular/common';

@Component({
    selector: 'app-status',
    imports: [NgClass],
    templateUrl: './status.component.html',
})
export class StatusComponent {
    public vm = input.required<StatusVM>();
}
