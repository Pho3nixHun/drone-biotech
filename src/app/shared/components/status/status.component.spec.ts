import { ComponentFixture, TestBed } from '@angular/core/testing';

import { StatusComponent } from './status.component';
import { Component } from '@angular/core';

@Component({
    imports: [StatusComponent],
    template: ` <app-status>Status</app-status> `,
})
class TestHostComponent {}

describe('StatusComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    //Snapshot test
    it('should render the template', () => {
        // Arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
