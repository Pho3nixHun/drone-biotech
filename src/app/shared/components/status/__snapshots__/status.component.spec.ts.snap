// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StatusComponent should render the template 1`] = `
<div
  id="root0"
  ng-version="20.1.7"
>
  <app-status>
    <p
      class="font-medium rounded-full w-max px-3 py-1 flex items-center text-blue-500"
    >
      Status
    </p>
  </app-status>
</div>
`;
